### PR TITLE
Add git commit and build timestamp to docker image tag, #66 and #58.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
script for windows will be update later.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: initializr-service/deploy-initializr-service.sh [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
 cd ../ && mvn clean package -Dmaven.test.skip=true -Pfull && cd -
-docker push ${your-acr-username}/azure-spring-cloud:latest && mvn azure-webapp:deploy
+
+image_info=($(docker images | grep -v "REPOSITORY" | head -1))
```
Comment (sophiaso) [inline initializr-service/deploy-initializr-service.sh]: Will this be very easy to be broken if another image built concurrently?

File: initializr-service/pom.xml [hunk 2]
```diff
@@ -12,6 +12,11 @@
 
 	<properties>
 		<main.basedir>${basedir}/..</main.basedir>
+		<maven.build.timestamp.format>yyyy_MM_dd-HH_mm</maven.build.timestamp.format>
+		<docker.image.prefix>${your-docker-image-prefix}</docker.image.prefix>
+        <docker.image.name>${your-docker-image-name}</docker.image.name>
```
Comment (sophiaso) [inline initializr-service/pom.xml:17]: Where are these `${your-docker-image-prefix} and ${your-docker-image-name}` and how to use them?

File: initializr-service/deploy-initializr-service.sh (hunk 1 above)
Comment (Incarnation-p-lee) [reply initializr-service/deploy-initializr-service.sh]: yeah, I am trying to push the image from dockfile, but always failure for now. Just workaround here.

File: initializr-service/pom.xml (hunk 2 above)
Comment (Incarnation-p-lee) [reply initializr-service/pom.xml:17]: from the doc, will guide you to finish this configuration.

Comment (Incarnation-p-lee) [reply initializr-service/pom.xml:17]: https://github.com/yaweiw/azure-spring-cloud-playground/blob/master/DEPLOY_TO_AZURE.adoc

File: initializr-service/deploy-initializr-service.sh (hunk 1 above)
Comment (Incarnation-p-lee) [reply initializr-service/deploy-initializr-service.sh]: Sorry, from dockfile-maven-plugin.

File: initializr-service/pom.xml (hunk 2 above)
Comment (sophiaso) [reply initializr-service/pom.xml:17]: Did not find the variables above in the doc though. Wonder can we use fixed image prefix and image name here to reduce customization?

Comment (Incarnation-p-lee) [reply initializr-service/pom.xml:17]: The doc should be updated due to pom change. The prefix related to ACR repo, cannot be fixed from azure portal.

Comment (yaweiw) [reply initializr-service/pom.xml:17]: Agree with @sophiaso. ${your-docker-image-prefix} and ${your-docker-image-name} are misleading. What do they refer to? e.g. 
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
demo/cloud-hystrix-dashboard   latest              3ac04a16ab90        3 days ago          132MB

Comment (yaweiw) [reply initializr-service/pom.xml:17]: ok then. Send PR to change Doc and give an example of what they refer to.

Comment (Incarnation-p-lee) [reply initializr-service/pom.xml:17]: This is not only related to the output from `docker images`, but more important for docker distributed images. How can we identify the unique docker images? It is clear if you consider that all docker images compose of `$prefix/$name:$tag`, `$prefix` can be dockerhub or ACR, `$name` is the docker image name.

**Review by yaweiw — APPROVED** [review]

**Pull request merged**